src/test/java/org/apache/maven/shared/filtering/PropertyUtilsTest.java

```diff
@@ -39,40 +38,40 @@ public class PropertyUtilsTest extends TestSupport {
     private static File testDirectory = new File(getBasedir(), "target/test-classes/");
 
     public void testBasic() throws Exception {
-        File basicProp = new File(testDirectory, "basic.properties");
+        File basicProp = File.createTempFile("basic", ".properties");
 
-        if (basicProp.exists()) {
+        try {
+            try (Writer writer = new OutputStreamWriter(new FileOutputStream(basicProp), StandardCharsets.UTF_8)) {
+                writer.write("ghost=${non_existent}\n");
+                writer.write("key=${untat_na_damgo}\n");
+                writer.write("untat_na_damgo=gani_man\n");
+                writer.flush();
+            }
+
+            Logger logger = mock(Logger.class);
+            Properties prop = PropertyUtils.loadPropertyFile(basicProp, false, false, logger);
+            assertTrue(prop.getProperty("key").equals("gani_man"));
+            assertTrue(prop.getProperty("ghost").equals("${non_existent}"));
+        } finally {
             basicProp.delete();
         }
-
-        basicProp.createNewFile();
-        try (FileWriter writer = new FileWriter(basicProp)) {
-            writer.write("ghost=${non_existent}\n");
-            writer.write("key=${untat_na_damgo}\n");
-            writer.write("untat_na_damgo=gani_man\n");
-            writer.flush();
-        }
-
-        Properties prop = PropertyUtils.loadPropertyFile(basicProp, false, false);
-        assertTrue(prop.getProperty("key").equals("gani_man"));
-        assertTrue(prop.getProperty("ghost").equals("${non_existent}"));
     }
 
     public void testSystemProperties() throws Exception {
-        File systemProp = new File(testDirectory, "system.properties");
+        File systemProp = File.createTempFile("system", ".properties");
 
-        if (systemProp.exists()) {
+        try {
+            try (Writer writer = new OutputStreamWriter(new FileOutputStream(systemProp), StandardCharsets.UTF_8)) {
+                writer.write("key=${user.dir}");
+                writer.flush();
+            }
+
+            Logger logger = mock(Logger.class);
+            Properties prop = PropertyUtils.loadPropertyFile(systemProp, false, true, logger);
+            assertTrue(prop.getProperty("key").equals(System.getProperty("user.dir")));
+        } finally {
```

Comment:
perhaps just set deleteOnExit instead to avoid IOExceptions here failing the test